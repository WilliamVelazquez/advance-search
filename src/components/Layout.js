import React from 'react';
import styled from 'styled-components';
import GlobalStyles from '../styled/GlobalStyles';
import Header from './Header';

import LOGO from '../assets/images/logo.svg';

const Content = styled.div`
  padding: 1em;
  padding-top: 110px;
`;

const Layout = (props) => {
  const { children } = props;
  return (
    <>
      <GlobalStyles />
      <Header logo={LOGO} logoAlt='Simetrik Logo' />
      <Content>
        {
          children
        }
      </Content>
    </>
  );
};

export default Layout;
