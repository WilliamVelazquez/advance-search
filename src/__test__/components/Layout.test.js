import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import 'jest-styled-components';

import Layout from '../../components/Layout';

describe('<Layout />', () => {
  const layout = mount(<Layout />);
  test('Rendering Layout Component', () => {
    expect(layout.length).toEqual(1);
  });
});

describe('Layout Snapshot', () => {
  test('Verifying Layout Component UI', () => {
    const layout = create(<Layout />);
    expect(layout.toJSON()).toMatchSnapshot();
  });
});
