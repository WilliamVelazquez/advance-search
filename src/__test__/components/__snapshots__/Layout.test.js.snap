// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Snapshot Verifying Layout Component UI 1`] = `
Array [
  .c0 {
  padding: 1em;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 3;
  background: #22242c;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 1000px;
  margin: 0 auto;
}

.c2 {
  margin: 0px;
  width: 130px;
  display: block;
}

.c3 {
  color: #3c75ff;
  cursor: pointer;
}

.c4 {
  width: 100%;
  padding-top: 10px;
}

@media (max-width:991px) {
  .c1 {
    width: 90%;
  }
}

@media (max-width:1140px) {
  .c1 {
    width: 960px;
  }
}

@media (max-width:1297px) {
  .c1 {
    width: 1180px;
  }
}

@media (max-width:1400px) {
  .c1 {
    width: 1200px;
  }
}

@media (max-width:1600px) {
  .c1 {
    width: 1300px;
  }
}

@media (max-width:1920px) {
  .c1 {
    width: 92%;
  }
}

@media (max-width:468px) {

}

@media (max-width:768px) {

}

@media (max-width:991px) {

}

<header
    className="c0"
  >
    <article
      className="c1"
    >
      <figure
        className="c2"
      >
        <a
          className="c3"
          href="/"
        >
          <img
            alt="Simetrik Logo"
            className="c4"
            src=""
          />
        </a>
      </figure>
    </article>
  </header>,
  .c0 {
  padding: 1em;
  padding-top: 110px;
}

@media (max-width:991px) {

}

@media (max-width:1140px) {

}

@media (max-width:1297px) {

}

@media (max-width:1400px) {

}

@media (max-width:1600px) {

}

@media (max-width:1920px) {

}

@media (max-width:468px) {

}

@media (max-width:768px) {

}

@media (max-width:991px) {

}

<div
    className="c0"
  />,
]
`;
